From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rainer Sigwald <raines@microsoft.com>
Date: Fri, 3 Dec 2021 14:24:03 -0600
Subject: [PATCH] Don't reference PublicApiAnalyzers in sourcebuild (#7118)

This package is only super relevant at PR-build time so it should
be fine to drop it from source-build scenarios. Fixes #7115.

Patch created from https://github.com/dotnet/msbuild/pull/7118
---
 src/Directory.Build.targets | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Directory.Build.targets b/src/Directory.Build.targets
index 2a840bfd7..bff6cc905 100644
--- a/src/Directory.Build.targets
+++ b/src/Directory.Build.targets
@@ -85,7 +85,7 @@
 
   <ItemGroup Condition="'$(GenerateReferenceAssemblySource)' == 'true'">
     <!-- Ensure API stability for shipping packages -->
-    <PackageReference Include="Microsoft.CodeAnalysis.PublicApiAnalyzers" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.CodeAnalysis.PublicApiAnalyzers" PrivateAssets="all" Condition="'$(DotNetBuildFromSource)' != 'true'" />
 
     <AdditionalFiles Include="PublicAPI/$(PublicApiTfm)/PublicAPI.Shipped.txt" />
     <AdditionalFiles Include="PublicAPI/$(PublicApiTfm)/PublicAPI.Unshipped.txt" />
